File: gitlfs-common/src/test/java/ru/bozaro/gitlfs/common/data/VerifyLocksReqTest.java
package ru.bozaro.gitlfs.common.data;

import org.testng.Assert;
import org.testng.annotations.Test;

import java.io.IOException;

public final class VerifyLocksReqTest {
  @Test
  public void parse01() throws IOException {
    final VerifyLocksReq data = SerializeTester.deserialize("verify-locks-req-01.json", VerifyLocksReq.class);
    Assert.assertNotNull(data);
    Assert.assertEquals(data.getCursor(), "optional cursor");
    Assert.assertEquals((Integer) 100, data.getLimit());
    Assert.assertNotNull(data.getRef());
    Assert.assertEquals(data.getRef().getName(), "refs/heads/my-feature");
  }
}

File: gitlfs-common/src/test/kotlin/ru/bozaro/gitlfs/common/data/VerifyLocksReqTest.kt
package ru.bozaro.gitlfs.common.data

import org.testng.Assert
import org.testng.annotations.Test
import java.io.IOException

class VerifyLocksReqTest {
    @Test
    @Throws(IOException::class)
    fun parse01() {
        val data: VerifyLocksReq = SerializeTester.deserialize("verify-locks-req-01.json", VerifyLocksReq::class.java)
        Assert.assertNotNull(data)
        Assert.assertEquals(data.cursor, "optional cursor")
        Assert.assertEquals(100, data.limit)
        Assert.assertNotNull(data.ref)
        Assert.assertEquals(data.ref!!.name, "refs/heads/my-feature")
    }
}
